resources/views/flights/edit.blade.php

```diff
@@ -0,0 +1,134 @@
+            const destinationSelect = document.getElementById('destination_select');
+
+            if (!airlineId) {
+                originSelect.classList.add('hidden');
+                destinationSelect.classList.add('hidden');
+            }
+
+            return axios.get(`/api/airlines/${airlineId}/cities`).then(res => {
+                cities = res.data.data;
+                const origin = document.getElementById('origin_city');
+                origin.innerHTML = '<option value="">Select Origin</option>';
+                cities.forEach(city => {
+                    const option = document.createElement('option');
+                    option.value = city.id;
+                    option.textContent = city.name;
+                    origin.appendChild(option);
+                });
+                updateDestinations();
+                originSelect.classList.remove('hidden');
+                destinationSelect.classList.remove('hidden');
+            });
+        }
+
+        function updateDestinations() {
+            const originId = document.getElementById('origin_city').value;
+            const destination = document.getElementById('destination_city');
+            destination.innerHTML = '<option value="">Select Destination</option>';
+            cities.filter(city => city.id != originId).forEach(city => {
+                const option = document.createElement('option');
+                option.value = city.id;
+                option.textContent = city.name;
+                destination.appendChild(option);
+            });
+        }
+
+        function loadFlightData() {
+            axios.get(`/api/flights/${flightId}`).then(res => {
+                const flight = res.data.data;
+                loadCitiesByAirline();
+            });
+        }
+
+        function updateFlight() {
+            const errorMessage = document.getElementById('flight_form_error');
+            const departureCity = document.getElementById('origin_city').value;
+            const arrivalCity = document.getElementById('destination_city').value;
+            const airline = document.getElementById('airline').value;
+            const departureTime = document.getElementById('departure_date').value;
+            const arrivalTime = document.getElementById('arrival_date').value;
+
+            errorMessage.textContent = '';
+
+            axios.put(`/api/flights/${flightId}`, {
+                departure_city_id: departureCity,
+                arrival_city_id: arrivalCity,
+                airline_id: airline,
+                departure_time: departureTime,
+                arrival_time: arrivalTime
+            })
+            .then(res => {
+                if (res.data.status === 'error') {
+                    errorMessage.textContent = res.data.message;
+                    return;
+                }
+                window.location.href = '/flights';
+            })
+            .catch(err => {
```

Comment:
you can change this from `err => { ...` to `{ response }` (if im not mistaken) so you can always use `response` instead of `err.response`